src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedObjectCreationTests.cs

```diff
@@ -4210,6 +4210,47 @@ public void M2(C c)
                 );
         }
 
+        [Fact]
+        public void TestInConstructorOverloadWithUseSiteError()
+        {
+            var missing = @"public class Missing { }";
+            var missingComp = CreateCompilation(missing, assemblyName: "missing");
+
+            var lib = @"
+public class C
+{
+    public C(Missing m) => throw null;
+    public C(D d) => throw null;
+}
+public class D { }
+";
+            var libComp = CreateCompilation(lib, references: new[] { missingComp.EmitToImageReference() });
+
+            var source = @"
+class D
+{
+    public void M2(C c)
+    {
+        new C(new());
```

Comment:
>new C [](start = 8, length = 5)

This is also not quite the scenario, the errors are not reported due to target typed new. We want the target typed new to be the only source of errors. Here we are mixing to much together.